### DISCUSSION TITLE
PR requires spec change.

### DISCUSSION THREAD

**Discussion by Lucipetus** [dlang/dmd#15086] [General]
I just opened a draft PR https://github.com/dlang/dmd/pull/15085.

But the tests failed because the problem I'm trying to solve is part of the spec:

>---
>        struct S(T) {}
>        alias A(T) = S!T;
>
>        static assert(is(A!int : S!T, T));
>        static assert(!is(A!int : A!T, T));
>        
>---
>__stdin.d(5): Error: static assert:  `!is(S!int : A!T, __isexp_id3, T)` is false

My PR is about making type deduction for Alias Template possible. It should be reviewed to see if it's plausible before the spec can be changed, but it can't even pass the tests because the spec forces the compiler not to do type deduction for Alias Template.

I don't know what to do now.

**Comment by salihdb** [comment]
It compiled for me with no errors.

**Reply by Lucipetus** [reply]
> It compiled for me with no errors.

That's sounds unlikely. My PR is exactly about resolving `is(A!int : A!T, T)` to be true. This code isn't supposed to compile.

**Comment by dkorpel** [comment]
There was already a PR open commenting out that line: https://github.com/dlang/dlang.org/pull/3571
I merged it.

**Reply by thewilsonator** [reply]
Thanks!
